### PR TITLE
update wtdbg2 to v2.3

### PR DESCRIPTION
**Pull request by mjsull**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by mjsull** [comment]
@bioconda I am getting a fatal error: zlib.h: No such file or directory, I've tried the suggestions in troubleshooting but it hasn't fixed the problem.

File: recipes/wtdbg/meta.yaml [hunk 1]
```diff
@@ -15,6 +15,8 @@ build:
 requirements:
   build:
     - {{ compiler('c') }}
+  host:
+    - zlib
```
Comment (thanhleviet) [inline recipes/wtdbg/meta.yaml]: You missed 
```
run:
- zlib 
```

File: recipes/wtdbg/build.sh [hunk 2]
```diff
@@ -2,5 +2,8 @@
 
 mkdir -p $PREFIX/bin
 
+export CFLAGS="-I$PREFIX/include"
+export LDFLAGS="-L$PREFIX/lib"
```
Comment (thanhleviet) [inline recipes/wtdbg/build.sh]: Replace line 5,6 with these:

export C_INCLUDE_PATH=${PREFIX}/include
export LIBRARY_PATH=${PREFIX}/lib

File: recipes/wtdbg/meta.yaml (hunk 1 above)
Comment (thanhleviet) [reply recipes/wtdbg/meta.yaml]: Ah no, you don't need it. Just replace 2 lines in build.sh as the tool does not use ./configure

**Comment by lh3** [comment]
Didn't notice this PR an hour earlier than my #13194. My PR has passed the linux test.

**Pull request closed without merging**